webroot/results/admin/persons_finish_unfinished.php

```diff
@@ -60,14 +60,18 @@ function getPersons () {
   global $personsFromPersons, $personsFromResultsWithoutId;
 
   $personsFromResultsWithoutId = [];
+  // when we are spliting wca profile, the person has an empty personId
+  // when we are uploading results, the person has an digit only personId
   $persons = dbQueryHandle("
-    SELECT DISTINCT personName, Results.countryId, year
+    SELECT personName, Results.countryId, year, Results.personId, Results.competitionId, InboxPersons.dob
     FROM Results
-    JOIN Competitions ON Competitions.id=Results.competitionId
-    WHERE personId=''
+    LEFT JOIN Competitions ON Competitions.id=Results.competitionId
+    LEFT JOIN InboxPersons ON InboxPersons.id=Results.personId and InboxPersons.competitionId=Results.competitionId
+    WHERE personId='' OR personId REGEXP '^[0-9]+$'
+    GROUP BY Results.personId, Results.personName, Results.competitionId
     ORDER BY personName
   ");
-  while( $row = mysql_fetch_row( $persons ))
+  while( $row = mysql_fetch_array( $persons ))
```

Comment:
Just curious: why the change here? I'm no PHP guy, but https://stackoverflow.com/a/11480149 says "It is recommended to use either _assoc or _row"?

Reply:
I don't know why they recommand use _assoc or _row instead of _array. In PHP doc it says:
>An important thing to note is that using mysql_fetch_array() is not significantly slower than using mysql_fetch_row(), while it provides a significant added value.